src/validation.jl

```diff
@@ -16,7 +16,12 @@ function check_method(@nospecialize(job::CompilerJob))
         cache = ci_cache(job)
         mt = method_table(job)
         interp = GPUInterpreter(cache, mt, job.source.world)
-        rt = Base.return_types(job.source.f, job.source.tt, interp)[1]
+        @static if VERSION ≥ v"1.8-beta2"
```

Comment:
Should we be more fine-grained here?

```suggestion
        @static if v"1.8-beta2" <= VERSION < v"1.9-" || VERSION ≥ v"1.9.0-DEV.190"
```

Reply:
Thanks, I just ignored the possibility to use `v"1.9-" < v"1.9.0-DEV.190"`.

Reply:
Yeah that makes sense, but it does happen that we need to bisect Julia in combination with GPUCompiler, so it's better to have the version checks be as accurate as possible then.